### PR TITLE
Remove attaching Satellite Infrastructure subscription

### PR DESCRIPTION
**Pull request by Lennonka**
#### What changes are you introducing?

Removal of attaching Satellite Infrastructure subscription

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

The procedure is only needed for SCA-disabled organizations, but in Foreman 3.12/Satellite 6.16, all organizations are SCA-enabled and SCA is enforced.

https://issues.redhat.com/browse/SAT-27606

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

I didn't find an include of this module in _Installing Server Disconnected_. Therefore, no removal there.

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [x] Foreman 3.11/Katello 4.13
* [x] Foreman 3.10/Katello 4.12
* [x] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

**Comment by Lennonka** [comment]
@evgeni Hi! Can you please review this PR? Should I also remove it from [System requirements](https://docs.theforeman.org/nightly/Installing_Server/index-satellite.html#system-requirements_satellite)?

**Comment by evgeni** [comment]
I am not exactly an expert in subs (:eyes: @jeremylenz @sideangleside), but will try my best here :)

This is what I know (or think to know, heh):
* Support for non-SCA manifests (= the subscriptions that are responsible for the content the Katello/Satellite *clients* receive) has been removed in Katello 4.12 (and thus any Satellite release based on this or later versions).
* Installing packages directly from cdn.redhat.com (after using `subscription-manager register` etc) is not affected by this, as that's a RHEL feature, not Katello/Satellite
* If I read https://access.redhat.com/articles/transition_of_subscription_services_to_the_hybrid_cloud_console correctly, *everyone* will be switched to SCA-mode in Oct/Nov, so you won't need `subscription-manager attach` to obtain specific subs anymore, even if you were using non-SCA mode against the CDN before.

For the docs/installation process, this means:
* You still need to *own* a Satellite Infra subscription (that's what "Your Satellite must have the Red Hat Satellite Infrastructure Subscription manifest in your Customer Portal" in the system requirements is about)
* You don't have to *attach* that subscription using `subscription-manager attach …`
* This applies to *all* Satellite versions, not only the ones that don't support non-SCA mode for *their* clients.

TL;DR:
* your patch is correct
* it needs to be cherry-picked to all supported releases, not only 4.14/6.16
* I am not sure whether this should happen now, or closer to the actual "force-SCA" date

**Comment by Lennonka** [comment]
Thank you. However, I would prefer additional ack from @jeremylenz. Are we sure about the cherry-picking to older Satellites?

**Comment by jeremylenz** [comment]
Right, if we're running `subscription-manager attach` on the Satellite and not on a host, then it doesn't matter whether the Satellite is SCA-only or not. I agree with @evgeni that this affects all Satellite versions, and we no longer need to attach the Satellite infrastructure subscription anywhere once all RH accounts are SCA.

As to the specific date that RH accounts will all be SCA, we can ask @sideangleside. :eyes:

**Comment by sideangleside** [comment]
The date when Red Hat accounts switch to SCA only is on/around 25-Oct-2024, as documented  [here](https://access.redhat.com/articles/transition_of_subscription_services_to_the_hybrid_cloud_console#important-upcoming-dates-and-releases-2) 

Also, `subscription-manager` will, in accounts where SCA is enabled, simply 'no-op' the request to attach. 

```
subscription-manager attach --pool "8a85f99c6895cc9901689a87d3f01323"
Ignoring the request to attach. Attaching subscriptions is disabled for organization "XXXXX" because Simple Content Access (SCA) is enabled.


```

**Comment by Lennonka** [comment]
Does it mean I should wait until Oct 25th to merge this PR?

**Comment by jeremylenz** [comment]
> * I am not sure whether this should happen now, or closer to the actual "force-SCA" date

My vote would be to just do it now. thoughts from others welcome

**Comment by Lennonka** [comment]
@ehelms Eric, can I merge and cherry pick this now?

**Comment by evgeni** [comment]
> > * I am not sure whether this should happen now, or closer to the actual "force-SCA" date
> 
> My vote would be to just do it now. thoughts from others welcome

So would be mine!

**Comment by Lennonka** [comment]
Alright then, let's party! :tada:

**Comment by Lennonka** [comment]
Cherry-picked:
- d6a76d1457..0ca4ae8495  3.12 -> 3.12
- 977bcad362..4bb5eb5e2f  3.11 -> 3.11
- 9b2c98d222..9ecce9d65c  3.10 -> 3.10
- 2c53c6e9f2..33751dd1c8  3.9 -> 3.9
- 3afdb52b6d..5eee4460bd  3.8 -> 3.8
- 6ac92302d5..ca9e615630  3.7 -> 3.7
- e45349c0be..a36828baaa  3.6 -> 3.6
- caa6138878..e1d652dcae  3.5 -> 3.5
